app/assets/javascripts/discourse/tests/integration/components/site-setting-test.js

```diff
@@ -59,4 +59,26 @@ module("Integration | Component | site-setting", function (hooks) {
 
     assert.strictEqual(query(".validation-error h1"), null);
   });
+
+  test("displays file types list setting", async function (assert) {
+    this.set("setting", {
+      setting: "theme_authorized_extensions",
+      value: "jpg|jpeg|png",
+      type: "file_types_list",
+    });
+
+    await render(hbs`<SiteSetting @setting={{this.setting}} />`);
+
+    assert.strictEqual(
+      query(".formatted-selection").innerText,
+      "jpg, jpeg, png"
+    );
+
+    await click(query(".file-types-list__button"));
```

Comment:
I think this should target a specific button (e.g. images) and also test that, if you click it again, the types are not doubled up (you have handled this case in the code, just need that extra test :) )